packages/components/src/components/hds/copy/snippet/index.ts

```diff
@@ -7,17 +7,47 @@ import Component from '@glimmer/component';
 import { assert } from '@ember/debug';
 import { tracked } from '@glimmer/tracking';
 import { action } from '@ember/object';
+import { HdsCopySnippetColorValues } from './types.ts';
+import type { HdsCopySnippetColors } from './types.ts';
+import type { HdsClipboardModifierSignature } from '../../../../modifiers/hds-clipboard.ts';
+
+export const DEFAULT_COLOR = HdsCopySnippetColorValues.Primary;
+export const COLORS: string[] = Object.values(HdsCopySnippetColorValues);
 
-export const DEFAULT_COLOR = 'primary';
-export const COLORS = ['primary', 'secondary'];
 export const DEFAULT_ICON = 'clipboard-copy';
 export const SUCCESS_ICON = 'clipboard-checked';
 export const ERROR_ICON = 'clipboard-x';
 export const DEFAULT_STATUS = 'idle';
 
-export default class HdsCopySnippetIndexComponent extends Component {
+interface HdsCopySnippetSignature {
+  Args: {
+    color?: HdsCopySnippetColors;
+    isFullWidth?: boolean;
+    isTruncated?: boolean;
+    onError?: HdsClipboardModifierSignature['Args']['Named']['onError'];
+    onSuccess?: HdsClipboardModifierSignature['Args']['Named']['onSuccess'];
+    textToCopy: HdsClipboardModifierSignature['Args']['Named']['text'];
```

Comment:
see: https://helios.hashicorp.design/components/copy/snippet?tab=code#component-api
```suggestion
    color?: HdsCopySnippetColors;
    isFullWidth?: boolean;
    textToCopy: HdsClipboardModifierSignature['Args']['Named']['text'];
    isTruncated?: boolean;
    onSuccess?: HdsClipboardModifierSignature['Args']['Named']['onSuccess'];
    onError?: HdsClipboardModifierSignature['Args']['Named']['onError'];
```